Randomizer/Patches/IPSPatcher.py

```diff
@@ -0,0 +1,45 @@
+from pathlib import Path
+from BaseClasses.PatchBase import *
+
+class IPSPatcher(PatchBase):
+    
+    def __init__(self, file, params = None):
+        super().__init__(file, params)        
+        
+    def __GetFile(self):
+        self.IPS_Folder = Path("IPSFiles/")
+        self.PatchFile = self.IPS_Folder / self.params
+        self.IPSFile = open(self.PatchFile, "r+b")
+    
+    def __Write(self):
+        Offset = 0
+        Size = 0
+        RLE_Size = 0
+        Value = 0
+        End_of_file_Checker = bytes(b'/x00')
+        x = 0
+        while End_of_file_Checker != bytes(b'EOF'):
+            Record = bytes(self.IPSFile.read(3))
+            End_of_file_Checker = Record
+            Offset = (int.from_bytes(Record, 'big')) - 0x10 #Subtracting the header size
```

Comment:
shouldnt that depend on weather the file is headered or not ?

Reply:
From my understanding all NES IPS files should assume a headered ROM, and all of our patching assumes an unheadered rom.

Something I learned recently is that NES headers tell the emulator what mapper/chipset that the ROM utilizes, as there's not a way to get this information from the ROM its self - It's part of emulating the chips inside a cartridge. What this means is, if a ROM files is unknown/modified it wont boot without a header. 

The second aspect is that IPS files can't write new information, they only overwrite what is already there. That means a IPS patch couldn't implement a header on an unheadered ROM. So for a patched ROM to boot, it needs to have a header already in place.

All of that to say, it wouldn't really make sense for an IPS file to expect an unheadered ROM. Of course we're still making assumptions 😅 but if someone made an IPS patch expecting an unheadered rom - that sounds more like an error in making the patch than us patching it incorrectly.

Of course I can still make changes if you'd like! Just wanted to share why I hardcoded it that way.